src/hackerspace_online/tests/test_middleware.py

```diff
@@ -0,0 +1,64 @@
+from io import BytesIO
+from unittest.mock import patch
+
+from django.test import TestCase, RequestFactory, override_settings
+from django.http import HttpResponse
+
+from hackerspace_online.middleware import RequestDataTooBigMiddleware
+
+
+def get_response_empty(request):
+    """Returns an empty response, for testing purpose"""
+    return HttpResponse()
+
+
+class RequestDataTooBigMiddlewareTestCase(TestCase):
+    """
+    Various tests for `hackerspace_online.middleware.RequestDataTooBigMiddleware` class
+    """
+
+    def setUp(self):
+        self.rf = RequestFactory()
+
+    def test_default(self):
+        """
+        By default request should pass through middleware as-is, without exceptions.
+        """
+        data = b"""{"orson": "wells"}"""  # request data
+
+        mw = RequestDataTooBigMiddleware(get_response_empty)
+        request = self.rf.request(
+            **{
+                "wsgi.input": BytesIO(data),
+                "CONTENT_LENGTH": len(data),
+            }
+        )
+
+        # should receive an empty response as if nothing has happened
+        response = mw(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.content, b"")
+
+    @override_settings(DATA_UPLOAD_MAX_MEMORY_SIZE=2, ALLOWED_HOSTS="*")
+    @patch("hackerspace_online.middleware.messages.add_message")
+    def test_request_data_too_big_exception(self, mock_add_message):
```

Comment:
This is probably just my lack of understanding, but it's not clear to me that this test ensures 1. That the specific exception is thrown and 2. that the specific exception is caught?  Maybe this isn't necessary, but from the code it's not obvious the test is specific enough.  Maybe it just needs to check the assert_called for the the correct message?  What are your thoughts?

Reply:
please, check https://github.com/bytedeck/bytedeck/pull/1455/commits/ade3bac3a1ae640980aa9e4fab2ff1088a343e6e

1) first case aka default behaviour, if nothing happened (request didn't exceeded the limit) act normal and do nothing

2) second case when request exceeded the size limit, but there are no custom middlewares that can handle `RequestDataTooBig` exception, response will be error page (400 Bad Request).

3) third case when request exceeded the size, but this time with installed custom middleware that can handle exception gracefully, redirect back to requested page and show **nice** error message via `django.contrib.messages` framework.

Do you need further explanation?

Reply:
Thank you! My comment was specifically about the test.  Is there a reason the test doesn't check that the RequestDataTooBig exception is thrown? That was my concern here.  From your explanation, seems like it should be checking that the specific exception is thrown before it tests that it correctly redirects?

Reply:
You have a test that check's without the middleware.  That works for me!  Seems more complicated than just checking for the exception, haha, but does the job!

Reply:
to @tylerecouture the whole point of middleware is to catch and handle `RequestDataTooBig` exception, if you want to test for that exception you need:

1. Make sure middleware is disabled
2. Make sure you run tests with DEBUG set to **True**

For some reason you test suite runs with DEBUG set to False, possible solution:

1. Set DEBUG=True in settings, line